### PR TITLE
Fix SAS Creation + other issue

### PR DESCRIPTION
**Pull request by mohsha-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/zc_sas_service.go [hunk 1]
```diff
@@ -36,7 +48,7 @@ type BlobSASSignatureValues struct {
 func (v BlobSASSignatureValues) NewSASQueryParameters(sharedKeyCredential *SharedKeyCredential) (SASQueryParameters, error) {
 	resource := "c"
 	if sharedKeyCredential == nil {
-		return SASQueryParameters{}, fmt.Errorf("cannot sign SAS query without Shared Key Credential")
+		return SASQueryParameters{}, fmt.Errorf("cannot sign SAS query without StorageAccountCredential")
```
[the remark this answers is not in the document]
Comment (mohsha-msft) [reply sdk/storage/azblob/zc_sas_service.go]: Fixed.

File: sdk/storage/azblob/zc_sas_service.go [hunk 2]
```diff
@@ -78,56 +98,31 @@ func (v BlobSASSignatureValues) NewSASQueryParameters(sharedKeyCredential *Share
...
-		contentEncoding:    v.ContentEncoding,
-		contentLanguage:    v.ContentLanguage,
-		contentType:        v.ContentType,
-		snapshotTime:       v.SnapshotTime,
-	}
-
-	// TODO: Mohit come here and ask Adele's help in understanding this segment of code.
-	//if udc, ok := sharedKeyCredential; ok {
-	//	udk := udc.GetUDKParams()
+	//udk := sharedKeyCredential.getUDKParams()
```
[the remark this answers is not in the document]
Comment (mohsha-msft) [reply sdk/storage/azblob/zc_sas_service.go:101]: I'll add support for User Delegation Key in coming release so I've commented the code changes in SAS.

**Comment by mohsha-msft** [comment]
Added Changelog Entries.

File: sdk/storage/azblob/CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,9 @@
 * Updated to latest `azcore`.  Public surface area is unchanged.
 
 ### Bugs Fixed
```
Comment (jhendrixMSFT) [inline sdk/storage/azblob/CHANGELOG.md:10]: Is there a plan to add all the missing changelog entries from when the storage dev branch was merged to main?

File: sdk/storage/azblob/CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,9 @@
...
+* Fixed Issue #16816 : ContainerClient.GetSASToken doesn't allow list permission.
```
Comment (seankane-msft) [inline sdk/storage/azblob/CHANGELOG.md:11]: Two nits here: 
1. If you want to include the issue number, we should include a link in the format `[issue num] (link/to/issue)`
2. The code portions (ContainerClient.GetSASToken, etc.) should be enclosed in backticks (`)

Comment (mohsha-msft) [reply sdk/storage/azblob/CHANGELOG.md:11]: 1. GitHub automatically links issue if given in the above format.

2. Yes, I'll add back-ticks to enclose code. Thanks for pointing this out.

File: sdk/storage/azblob/CHANGELOG.md (hunk 3 above)
Comment (mohsha-msft) [reply sdk/storage/azblob/CHANGELOG.md:10]: Yes, I have plans to add more changelogs. There is one more bug which I am planning to solve after which I'll do the next azblob release.

**Pull request merged**